src/error.jl

```diff
@@ -27,16 +25,62 @@ end
 function gdaljl_errorhandler(class::CPLErr, errno::Cint, errmsg::Cstring)
     # function signature needs to match the one in __init__, and the signature
     # of the callback for a custom error handler in the GDAL docs
-    if class in throw_class
+    if class === CE_Failure
         throw(GDALError(class, errno, unsafe_string(errmsg)))
     end
     return C_NULL
 end
 
-"Throw an error if a pointer is null and GDAL reports an error"
-function failsafe(ptr)
-    if ptr == C_NULL && cplgetlasterrortype() in throw_class
+"Check the last error type and throw a GDALError if it is a failure"
+function maybe_throw()
```

Comment:
This function is oddly familiar to the above `gdaljl_errorhandler`. Is the above still used/needed?
The name seems too generic too, it could be something like `potential_errorhandler` or similar?

Reply:
Hmm `maybe_throw` is not an error handler though. The `gdaljl_errorhandler` is still used. If GDAL itself throws an error, `gdaljl_errorhandler` will catch it and throw a `GDALError`. `maybe_throw` is instead used from our side, and the only thing it does is maybe throw a `GDALError` if there is a `CE_Failure`.

Reply:
Ah, handling C libraries is not easy. Would be nice to have your comment (about who throws what, calling which function) in the code here.

Reply:
9f9a754f86c0b92f511c78ac40690b0a29390744